### PR TITLE
Mb temp sensor

### PR DESCRIPTION
**Pull request by epeach**
# Description
Enables the temperature sensor on the MicroBit.

tempSensor

## Links

- [spec]()
- [jira]()

## Testing story
Testing coming after demo

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/kits/maker/Thermometer.js [hunk 1]
```diff
@@ -31,4 +32,60 @@ const Thermometer = {
...
+    super();
+    this.currentTemp = 0;
+    this.board = board;
+    this.board.mb.addFirmataUpdateListener(() => {
+      this.emit('data');
+
+      // If the temp value has changed, update the local value and
+      // trigger a change event
+      if (this.currentTemp !== this.board.mb.analogChannel[12]) {
+        this.currentTemp = this.board.mb.analogChannel[12];
```
Comment (jmkulwik) [inline apps/src/lib/kits/maker/Thermometer.js]: Mind adding this channel ID to a constants file?

File: apps/src/lib/kits/maker/Thermometer.js [hunk 2]
```diff
@@ -31,4 +32,60 @@ const Thermometer = {
...
+      },
+      celsius: {
+        get: function() {
+          return this.raw;
+        }
+      },
+      fahrenheit: {
+        get: function() {
+          let rawValue = (this.celsius * 9) / 5 + 32;
+          return Math.floor(rawValue * 100) / 100;
```
Comment (jmkulwik) [inline apps/src/lib/kits/maker/Thermometer.js]: I noticed you used a `roundToHundreth` function in your Accelerometer PR. Seems like a handy function for a lot of MicroBit work. Mind making it a util function and using it here too?

File: apps/src/lib/kits/maker/Thermometer.js [hunk 3]
```diff
@@ -31,4 +32,60 @@ const Thermometer = {
     }
   }
 };
-export default Thermometer;
+export default PlaygroundThermometer;
+
+export class MicroBitThermometer extends EventEmitter {
+  constructor(board) {
+    super();
+    this.currentTemp = 0;
```
Comment (jmkulwik) [inline apps/src/lib/kits/maker/Thermometer.js:41]: Talked offline. This causes the initial reading to be 0C the first time the program is run after loading the page

Comment (epeach) [reply apps/src/lib/kits/maker/Thermometer.js:41]: Coming back to this - this line isn't what is causing the initial reading to be 0C. this.currentTemp is only used to determine whether a data event can trigger the change event (i.e. has the temp changed from what it was before), but only the sensor value is ever returned from the thermometer. This issue is happening because analog channels are initialized to 0 and then we are reading the channel value before any data events have been emitted. So receiving the 0C here is behaving as expected (i.e. sensor hasn't received data, so it returns the initialized value).

tldr: Behavior is expected. No changes needed.

**Comment by epeach** [comment]
@jmkulwik PTAL!

File: apps/src/lib/kits/maker/Thermometer.js (hunk 3 above)
Comment (jmkulwik) [reply apps/src/lib/kits/maker/Thermometer.js:41]: Hmm that feels like a negative user experience - i.e. the first time I (or my friends/family/teacher) uses my app, it will always show 0C. Is there a way we can wait until the data event has been emitted to show temperature?

Comment (jmkulwik) [reply apps/src/lib/kits/maker/Thermometer.js:41]: Synced offline. Not needed to change this for the BETT demo. Erin's tracking this in follow-up work.

**Review by jmkulwik — APPROVED** [review]
LGTM!

File: apps/src/lib/kits/maker/Thermometer.js (hunk 3 above)
Comment (epeach) [reply apps/src/lib/kits/maker/Thermometer.js:41]: Jira ticket to track^ https://codedotorg.atlassian.net/browse/STAR-936
